src/celeritas/em/interactor/WentzelInteractor.hh

```diff
@@ -0,0 +1,159 @@
+    ParticleTrackView const& particle_;
+
+    // Target isotope
+    IsotopeView const& target_;
+
+    // Scattering direction distribution of the Wentzel model
+    WentzelDistribution const sample_direction;
+
+    //// HELPER FUNCTIONS ////
+
+    //! Calculates the recoil energy for the given scattering direction
+    inline CELER_FUNCTION real_type calc_recoil_energy(
+        Real3 const& new_direction, Mass const& target_mass) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from shared and state data
+ */
+CELER_FUNCTION
+WentzelInteractor::WentzelInteractor(WentzelRef const& shared,
+                                     ParticleTrackView const& particle,
+                                     Real3 const& inc_direction,
+                                     IsotopeView const& target,
+                                     ElementId const& el_id,
+                                     CutoffView const& cutoffs)
+    : inc_direction_(inc_direction)
+    , particle_(particle)
+    , target_(target)
+    , sample_direction(particle,
+                       target,
+                       shared.elem_data[el_id],
+                       value_as<Energy>(cutoffs.energy(particle.particle_id())),
+                       shared)
+{
+    CELER_EXPECT(particle_.particle_id() == shared.ids.electron
+                 || particle_.particle_id() == shared.ids.positron);
+    CELER_EXPECT(particle_.energy() > detail::coulomb_scattering_limit()
+                 && particle_.energy() < detail::high_energy_limit());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the Coulomb scattering of the incident particle.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction WentzelInteractor::operator()(Engine& rng)
+{
+    // Incident particle scatters
+    Interaction result;
+
+    // Sample the new direction
+    Real3 new_direction = sample_direction(rng);
+    result.direction = rotate(inc_direction_, new_direction);
+
+    // Recoil energy is kinetic energy transfered to the atom
+    real_type inc_energy = value_as<Energy>(particle_.energy());
+    real_type recoil_energy
+        = calc_recoil_energy(new_direction, target_.nuclear_mass());
+    CELER_EXPECT(0 <= recoil_energy && recoil_energy <= inc_energy);
+    result.energy = Energy{inc_energy - recoil_energy};
+
+    // TODO: For high enough recoil energies, ions are produced
+
+    result.energy_deposition = Energy{recoil_energy};
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the recoil energy for the given scattering direction calculated
+ * by WentzelDistribution.
+ */
+CELER_FUNCTION real_type WentzelInteractor::calc_recoil_energy(
+    Real3 const& new_direction, Mass const& target_mass) const
```

Comment:
Now that `target_mass` is constant over the lifetime of the sampler (since the isotope has already been selected) you can remove the argument and replace it with `target_.nuclear_mass()`.